dask/array/rechunk.py

```diff
@@ -100,10 +102,19 @@ def _intersect_1d(breaks):
         last_end = end
         if br == last_br:
             if label == "o":
+                last_o_br = br
+                last_o_end = end
                 old_idx += 1
+            if label == "n" and last_label == "n":
+                if last_o_br == br and old_idx > 0:
+                    ret_next.append((old_idx - 1, slice(last_o_end, last_o_end)))
+                else:
+                    ret_next.append((old_idx, slice(start, end)))
```

Comment:
I recognize that most of this code is uncommented, but would you mind adding a small comment here (and maybe other places in this function as you see fit) on what each branch appending to `ret_next` signifies? Would make this code more approachable for the next contributor.

Reply:
Sure.  No problem.